In [1]:
!pip install statsmodels

In [2]:
#importing libraries
import pandas as pd
import numpy
import statsmodels.api as sm
import statsmodels.formula.api as smf
import re


In [ ]:
data = pd.read_csv('data.csv') #importing data

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43093 entries, 0 to 43092
Columns: 3008 entries, ETHRACE2A to OTHBP12ABDEP
dtypes: float64(11), int64(654), object(2343)
memory usage: 989.0+ MB


In [ ]:
#making numeric
data['S3AQ3D1R'] = pd.to_numeric(data['S3AQ3D1R'], errors='coerce')
data['S10Q1A50'] = pd.to_numeric(data['S10Q1A50'], errors='coerce')

In [7]:
data = data[data['S10Q1A50'] != 9]
data['S10Q1A50'] = data['S10Q1A50'] - 1
#avoiding n/a and making binar scale to zero

<ipython-input-7-eff89d6fc9b0>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['S10Q1A50'] = data['S10Q1A50'] - 1


In [8]:
from collections import Counter
Counter(data['S10Q1A50'].to_list())

Counter({1: 37850, 0: 3520})

In [9]:
answer = smf.ols('S3AQ3D1R ~ S10Q1A50', data=data).fit()
print('OLS regression model for assosiation between DURATION (DAYS) OF USUAL CIGARETTE SMOKING and the agreement to VERY FEW THINGS THAT GIVE YOU PLEASURE ', answer)

OLS regression model for assosiation between DURATION (DAYS) OF USUAL CIGARETTE SMOKING and the agreement to VERY FEW THINGS THAT GIVE YOU PLEASURE  <statsmodels.regression.linear_model.RegressionResultsWrapper object at 0x0000021BCA9F5490>


In [12]:
answer.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               S3AQ3D1R   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     8.196
Date:                Thu, 17 Mar 2022   Prob (F-statistic):            0.00420
Time:                        22:43:04   Log-Likelihood:            -2.0741e+05
No. Observations:               17411   AIC:                         4.148e+05
Df Residuals:                   17409   BIC:                         4.148e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    8.64e+04    881.779     97.981      0.000    8.47e+04    8.81e+04
S10Q1A50   -2655.2786    927.493     -2.863      0.004   -4473.257    -837.300
==============================================================================
Omnibus:                     5068.499   Durbin-Watson:                   1.977
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            10782.837
Skew:                          -1.814   Prob(JB):                         0.00
Kurtosis:                       4.304   Cond. No.                         6.30
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

# Summary
## Statistics
### P>|t| = 0.000, 0.004, intercept coef = , 'yes' answer to 'VERY FEW THINGS THAT GIVE YOU PLEASURE' = -2655.2786
## Conclusion
### Results statistically significant. Correlation is negative.